### ISSUE TITLE
Add filters for registered post types and registered taxonomy connections.

### ISSUE THREAD

**Issue by BE-Webdesign** [wp-graphql/wp-graphql#132]
Currently post types and taxonomies that are registered to support a specific subset of taxonomies are not being fully respected by the plugin.  So when a post type like pages which doesn't support categories or tags is queried against the collection, only empty edges will be found even if pages have been assigned a category via `wp_set_object_terms()`.  We should change `$allowed_post_types` and `$allowed_taxonomies` to be a filterable value and more specific to each taxonomy or post type.  We can have all of the values stay the way they are right now, but we can add filters that will take a subset of those allowed types and which will remove and fields on the taxonomy/post type that will go no where.

**Comment by jasonbahl** [comment]
@BE-Webdesign I'm not sure I follow completely. 

Are you saying that the categoriesEdge should only contain queries for postTypes that were supported? For example, this should not be a valid query, because the "page" post_type isn't connected to the "category" taxonomy?

```
query{
  categories{
    edges{
      node{
        name
        taxonomy
        pages{
          edges{
            node{
              id
              title
            }
          }
        }
      }
    }
  }
}
```
I think this: https://github.com/wp-graphql/wp-graphql/blob/master/src/Type/TermObject/TermObjectType.php#L188-L193

Probably just needs to get updated to be more like this: 
https://github.com/wp-graphql/wp-graphql/blob/master/src/Type/PostObject/PostObjectType.php#L354-L362|

Where we check to make sure the Taxonomies / Post Types are both "allowed" AND registered as connections. 

So, right now the above query shouldn't even be allowed, because pages aren't a registered connection to the category taxonomy, so the categoriesEdge should only have `posts` as the only postObjectConnection out of the box. 

Am I on the same page as you now?

**Comment by BE-Webdesign** [comment]
I think so, sorry for my horrible explanation lol, it was hard to put into words.

**Comment by BE-Webdesign** [comment]
So on post types it is good just the other way we need to do the same thing.

**Comment by BE-Webdesign** [comment]
I was thinking we could include a filter as well but on second thought it is not necessary as every field is already filtered.

**Comment by jasonbahl** [comment]
Cool. I'll do a patch to implement my suggested fix in a bit. Thanks for catching and pointing that out!

**Comment by BE-Webdesign** [comment]
Yup, if we merged the taxonomy stuff already, I can't remember you should get some failing tests too!

**Comment by jasonbahl** [comment]
This was fixed, I believe by #174
